File: app/src/main/java/com/example/weizhenbin/wangebug/tools/PhoneTool.java
package com.example.weizhenbin.wangebug.tools;

import com.example.weizhenbin.wangebug.base.App;

/**
 * Created by weizhenbin on 2018/8/21.
 */

public class PhoneTool {

    public static  int getScreenHeight(){
        return App.Companion.getApp().getResources().getDisplayMetrics().heightPixels;
    }
    public static  int getScreenWidth(){
        return App.Companion.getApp().getResources().getDisplayMetrics().widthPixels;
    }


    public static int dip2px(float dpValue) {

        final float scale = App.Companion.getApp().getResources().getDisplayMetrics().density;

        return (int) (dpValue * scale +0.5f);

    }

    public static int px2dip(float pxValue) {

        final float scale = App.Companion.getApp().getResources().getDisplayMetrics().density;

        return (int) (pxValue / scale +0.5f);

    }


}

File: app/src/main/java/com/example/weizhenbin/wangebug/tools/PhoneTool.kt
package com.example.weizhenbin.wangebug.tools

import com.example.weizhenbin.wangebug.base.App

/**
 * Created by weizhenbin on 2018/8/21.
 */

object PhoneTool {

    val screenHeight: Int
        get() = App.app.resources.displayMetrics.heightPixels
    val screenWidth: Int
        get() = App.app.resources.displayMetrics.widthPixels


    fun dip2px(dpValue: Float): Int {

        val scale = App.app.resources.displayMetrics.density

        return (dpValue * scale + 0.5f).toInt()

    }

    fun px2dip(pxValue: Float): Int {

        val scale = App.app.resources.displayMetrics.density

        return (pxValue / scale + 0.5f).toInt()

    }


}
